baddns/lib/httpmanager.py

```diff
@@ -22,26 +26,43 @@ def __init__(self, target, http_client_class=None, skip_redirects=False):
             setattr(self, attr, None)
 
     async def dispatchHttp(self):
-        try:
-            protocols = ["http", "https"]
-            for protocol in protocols:
-                base_url = f"{protocol}://{self.target}/"
-                # If redirects are not skipped by this module, perform requests that follow redirects
-                if not self.skip_redirects:
-                    setattr(
-                        self,
-                        f"{protocol}_allowredirects_results",
-                        await self.http_client.get(base_url, follow_redirects=True),
-                    )
-                # Always perform requests that do not follow redirects
-                setattr(
-                    self,
-                    f"{protocol}_denyredirects_results",
-                    await self.http_client.get(base_url, follow_redirects=False),
-                )
-        except (httpx.RequestError, httpx.ConnectError, httpx.TimeoutException) as e:
-            log.debug(f"An httpx error occurred while requesting {e.request.url!r}: {e}")
-        except ssl.SSLError as e:
-            log.debug(f"An SSL error occurred while requesting {e}")
-        except anyio.EndOfStream as e:
-            log.debug(f"An anyio error occurred while requesting {e}")
+        protocols = ["http", "https"]
+        tasks = []
+
+        for protocol in protocols:
+            base_url = f"{protocol}://{self.target}/"
+            log.debug(f"ready to make request to URL: {base_url}")
+
+            if not self.skip_redirects:
+                tasks.append((self.http_client.get(base_url, follow_redirects=True), base_url))
+
+            tasks.append((self.http_client.get(base_url, follow_redirects=False), base_url))
+
+        tasks_with_urls = [(task, url) for task, url in tasks]
```

Comment:
tasks_with_urls is identical to tasks?

Reply:
tasks is just a list of the task, the tasks_with_urls list is a (task,url) tuple used for deconflicting after they are all done

Reply:
🤔

Reply:
i updated this to make it less redundant. I think there was some other reason i did that, and ended up removing leaving the redundant generator left over.